### DISCUSSION TITLE
I ran 30 cameras on Proxmox as a VM using Ubuntu Noble LTS

### DISCUSSION THREAD

**Discussion by Taiga25-byte** [blakeblackshear/frigate#13414] [General]
Hey All,

I'm totally new to NVRs like frigate with little to no experience with self hosting.
I recently discovered this and would like to start using to for our small scale business which we have around 35 cameras with a mix of Sony, Hikvision and Panasonic.
My Ubuntu VM on Proxmox are as follows:
CPU: 8 cores 3 threads (Base: Intel Xeon)
Memory: 64GB
Storage: 4TB
GPU: VirtIO-GPU
I have no TPU inserted into the server.
I have no issue with receiving the streams from the cameras using RTSP, I disabled the detection as I don't need that but I have some issues with recording. The streams would record for an hour then stopped but I can still see the live view. 
And when I use the live view, sometimes the streams would froze and I have to refresh the webpage.
I don't see any specific errors in the docker logs output and have no clue what I am doing wrong.
Any advices would be greatly appreciated.

Docker Compose File
```
#version: "27.1.2"
services:
  frigate:
    container_name: frigate
    restart: unless-stopped
    image: [ghcr.io/blakeblackshear/frigate:stable](http://ghcr.io/blakeblackshear/frigate:stable)
    shm_size: "1000mb"
    volumes:
      - ./config:/config
      - ./storage:/media/frigate
      - type: tmpfs # Optional: 1GB of memory, reduces SSD/SD Card wear
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - "8971:8971"
      - "8554:8554" # RTSP feeds
```
Config.yml
```
mqtt:
  enabled: false

ui:
  time_format: browser
  date_style: short
  time_style: medium
  strftime_fmt: "%Y/%m/%d %H:%M"

go2rtc:
  streams:
    Sony:
      - rtsp://username:password@IP:554/H.264

record:
  enabled: true
  retain:
    days: 30
  events:
    retain:
      default: 1

cameras:
 Hikvision: # <------ Name the camera
    enabled: true
    ffmpeg:
      inputs:
        - path: rtsp://username:password@IP:554/H.264 # <----- The stream you want to use for detection
          roles:
            - record
    detect:
      enabled: false # <---- disable detection until you have a working camera feed
      width: 1280
      height: 720
version: 0.14
```

**Comment by NickM-27** [comment]
We will need to see the logs

**Reply by Taiga25-byte** [reply]
2024-08-29 14:03:10.358588697  [2024-08-29 14:03:10] frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for Hikvision. Keeping the 6 most recent segments out of 7 and discarding the rest...

I saw these lines for all cameras. I'm not sure what it means and which area for me to look.

**Reply by NickM-27** [reply]
We'll need to see a full copy of logs

**Reply by Taiga25-byte** [reply]
Hello Nick,
I tried to increase the size of /dev/shm and the recording seems to working fine for a few days but I saw some gaps randomly over the recording period. 
Let me rename the cameras and share the logs with you.

**Reply by Taiga25-byte** [reply]
@NickM-27 
Hello Nick, 
After running it and monitoring it for a while, I saw that the recordings stop from time to time and the most recent one is today. 
I was going through the logs and found this below at the timestamp where CPU usage drops and recordings stopped.
Could you review it and let me know what might be the issue?
[Record_Stopped_Logs.txt](https://github.com/user-attachments/files/16893664/Record_Stopped_Logs.txt)
Thank you so much for your time.

**Reply by NickM-27** [reply]
yes, this is the SHM issue, you can try running the latest dev image which has rewritten the way that SHM works to avoid issues like this
